components/Util.js

```diff
@@ -72,78 +69,166 @@ export const generateRandVals = async (size) => {
+    const [, identifier, operator] = key
+      .split("[")
+      .map((i) => i.replace("]", ""));
     return {
       value,
       operator,
-      identifier
+      identifier,
     };
-  })
- )
-  
-const convertToNumber = items => {
-  const converted = items.map(item => Number(item));
+  });
+
+const convertToNumber = (items) => {
+  const converted = items.map((item) => Number(item));
   return converted.includes(NaN) ? false : converted;
-}; 
+};
+
+export const sortData = (data, sortValue) => {
+  const splitted = sortValue.split(":");
+  const [identifier, order] = splitted;
+
+  let sortedData = Object.values(data).sort((a, b) => {
+    if (a[identifier] < b[identifier]) {
+      return -1;
+    }
+    if (a[identifier] > b[identifier]) {
+      return 1;
+    }
+    return 0;
+  });
+
+  if (order?.toLowerCase() === "desc") {
+    sortedData = sortedData.reverse();
+  }
+
+  return sortedData;
+};
+
+const substringSearch = (item) => {
+  const values = item.match(/\((.*?)\)/)[1].split(", ");
+  const converted = values
+    .map((value) => `slug LIKE '%${value}%'`)
+    .join(" OR ");
+  return converted;
+};
+
+const valueTypesIterator = (operator, value, isSubstringSearch = false) => {
+  const valueTypes = ["value_string", "value_long_string", "value_double"];
+  const isNotCovertible = isNumber(value);
+
+  const convertedValue = isNotCovertible ? value : `"${value}"`;
+  const finalValue = isNotCovertible ? convertedValue : `(${convertedValue})`;
+
+  let combinedValues;
+  if (!isNotCovertible) {
+    combinedValues = valueTypes.map((columnName, index) => {
+      return `${columnName} ${operator} ${
+        !isSubstringSearch ? `${finalValue}` : `"%${value}%"`
+      }${valueTypes.length != index + 1 ? " OR" : ""}`;
+    });
+  } else {
+    combinedValues = `value_double ${operator} ${
+      !isSubstringSearch ? `${finalValue}` : `"%${value}%"`
+    }`;
+    // only return long_double
+  }
+
+  return isNotCovertible ? combinedValues : `${combinedValues.join(" ")}`;
+};
 
-const isGreaterThan = (condition,value,orEqual=false) =>{
-  const final = condition.map((eachCon)=>{
-   return orEqual ? eachCon < value ? true : false : eachCon <= value ? true : false 
-})
-  return final.every(item => item === true);
+export const findCommonNumbers = (rawIds) => {
+
+  const data = rawIds.map(obj => obj.records)
+  const firstVal = data.filter((arr,i) => i === 0).map(arr =>
+    arr.map(([{longValue: num}]) => num))
+    
+    const compareValue = data.filter((arr,i) => i !== 0).map(arr =>
+    arr.map(([{longValue: num}]) => num))
+  
+  const intersection = firstVal[0].filter((num1, index, arr1) => {
+    return compareValue.every(arr2 => {
+      return arr2.includes(num1);
+    }) && arr1.indexOf(num1) === index;
+  });
+       
+  const idsCondition = `(${intersection.map(String).join(",")})`;
+  return  intersection.length !== 0 ? idsCondition : false
```

Comment:
This is essentially doing set intersection in-memory, instead on the database. Our application would probably run out of memory on large sets, especially on our serverless application. 

We should refactor this to let the database do the intersection instead of the code.